src/command/restore/restore.c

```diff
@@ -2473,6 +2502,12 @@ cmdRestore(void)
             strNewFmt(STORAGE_REPO_BACKUP "/%s/" BACKUP_MANIFEST_FILE, strZ(backupData.backupSet)), backupData.repoCipherType,
             backupData.backupCipherPass);
 
+        if (cfgOptionTest(cfgOptFilter) &&
+            (cfgOptionStrId(cfgOptFork) != CFGOPTVAL_FORK_GPDB || manifestData(jobData.manifest)->pgVersion != PG_VERSION_94))
+        {
+            THROW(OptionInvalidError, "option '" CFGOPT_FILTER "' is supported on GPDB 6 only");
+        }
```

Comment:
may be something like
```suggestion
        if (cfgOptionTest(cfgOptFilter) &&
            (cfgOptionStrId(cfgOptFork) != CFGOPTVAL_FORK_GPDB || manifestData(jobData.manifest)->pgVersion != PG_VERSION_94))
        {
            if (cfgOptionStrId(cfgOptFork) != CFGOPTVAL_FORK_GPDB)
                THROW(OptionInvalidError, "option '" CFGOPT_FILTER "' is supported on GPDB only");
            if (manifestData(jobData.manifest)->pgVersion != PG_VERSION_94))
                THROW(OptionInvalidError, "option '" CFGOPT_FILTER "' is supported on GPDB 6 only");
        }
```

Reply:
I think that `GPDB 6 only` is suitable for the both cases and more informative. The code will be simpler: one THROW instead of two

Reply:
But to add support for the 7X, you will have to change less.